CargoMonitor/CargoMonitor.cs

```diff
@@ -953,15 +1000,19 @@ private void readInventory(CargoMonitorConfiguration configuration = null)
             }
         }
 
-        private void AddCargo(Cargo cargo)
+        private void AddOrUpdateCargo(Cargo cargo)
         {
-            if (cargo == null)
-            {
-                return;
-            }
-
+            if (cargo == null) { return; }
             lock (inventoryLock)
             {
+                for (int i = 0; i < inventory.Count; i++)
+                {
+                    if (string.Equals(inventory[i].edname, cargo.edname, StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        inventory[i] = cargo;
+                        return;
```

Comment:
A `return` this deep is always a bit of a loaded gun should the code be blindly moved into another context.

I would instead of the manual `for` loop use `inventory.FindIndex()` with a predicate and update if found or Add() if not.